packages/cli/src/checkBeforeRunning.ts

```diff
@@ -0,0 +1,16 @@
+import {Hook} from '@oclif/config'
+import {readFile} from 'fs-extra'
+
+const whitelistGlobal = ['new']
+
+const hook: Hook<'init'> = async function (options) {
+  if (options.id && whitelistGlobal.includes(options.id)) return
+
+  try {
+    await readFile('blitz.config.js')
+  } catch (err) {
+    this.error(`It appears that you're not running in a blitz project`)
+  }
```

Comment:
Ok few things:

1. `blitz.config.js` is optional, so this won't work. We do need robust logic for detecting if we are in a blitz app. Maybe if `blitz` is a dependency of the package.json? 
2. Put the above detection logic in it's own file so we can reuse elsewhere
3. If not at root of a blitz project, check if we are in a sub directory of a blitz project and have a specific error message for this.
4. Else have an error about not being in a blitz project.

Reply:
**Error message for in a sub-folder:** "You are currently in a sub-folder of your Blitz app, but this command must be used from the root of your project."

**Error message for not in a blitz app:** "You are not inside a Blitz project, so this command won't work. You can create a new app with `blitz new myapp` or see help with `blitz help`